cpp/src/io/statistics/byte_array_view.cuh

```diff
@@ -0,0 +1,130 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cudf/utilities/span.hpp>
+
+namespace cudf::io {
+
+class byte_array_view {
+ public:
+  using element_type = int8_t const;  ///< The type of the elements in the byte array
+
+  constexpr byte_array_view() noexcept {}
+  /**
+   * @brief Constructs a byte_array_view from a pointer and a size.
+   *
+   * @param data Pointer to the first element in the byte array.
+   * @param size The number of elements in the byte array.
+   */
+  constexpr byte_array_view(element_type* data, std::size_t size) : _data(data, size) {}
+  constexpr byte_array_view(byte_array_view const&) noexcept = default;  ///< Copy constructor
+  /**
+   * @brief Copy assignment operator.
+   *
+   * @return Reference to this byte_array_view.
+   */
+  constexpr byte_array_view& operator=(byte_array_view const&) noexcept = default;
+
+  /**
+   * @brief Returns a reference to the idx-th element of the byte_array_view.
+   *
+   * The behavior is undefined if idx is out of range (i.e., if it is greater than or equal to
+   * size()).
+   *
+   * @param idx The index of the element to access.
+   * @return A reference to the idx-th element of the byte_array_view, i.e., `_data.data()[idx]`.
+   */
+  constexpr element_type& operator[](std::size_t idx) const { return _data[idx]; }
+
+  /**
+   * @brief Returns a pointer to the beginning of the byte_array_view.
+   *
+   * @return A pointer to the first element of the byte_array_view.
+   */
+  constexpr element_type* data() const noexcept { return _data.data(); }
+
+  /**
+   * @brief Returns the number of elements in the byte_array_view.
+   *
+   * @return The number of elements in the byte_array_view.
+   */
+  [[nodiscard]] constexpr std::size_t size() const noexcept { return _data.size(); }
+
+  /**
+   * @brief Returns the size of the byte_array_view in bytes.
+   *
+   * @return The size of the byte_array_view in bytes
+   */
+  [[nodiscard]] constexpr std::size_t size_bytes() const noexcept { return _data.size_bytes(); }
+
+  /**
+   * @brief Comparing target byte_array_view with this byte_array_view. Each byte in the array is
+   * compared.
+   *
+   * @param byte_array_view Target byte_array_view to compare with this byte_array_view.
+   * @return 0  If they compare equal.
+   *         <0 Either the value of the first byte of this byte_array_view that does not match is
+   * lower in the arg byte_array_view, or all compared bytes match but the arg byte_array_view is
+   * shorter. >0 Either the value of the first byte of this byte_array_view that does not match is
+   * greater in the arg byte_array_view, or all compared bytes match but the arg byte_array_view is
+   * longer.
+   */
+  __device__ inline int32_t compare(byte_array_view const& rhs) const
```

Comment:
It's a debate for sure. On one hand I miss out on things in `device_span` that people may want to use and I don't pick up new things for free, but on the other hand it is limited in scope and we can hopefully keep things simple.

Reply:
I guess the question that could affect the approach is "is byte_array_view a span?". If there is any aspect of span that does not apply to the view class, then composition is in order. IMO a big plus for inheritance is that it would melt away a lot of code.